lib/kcerts/ssh.go

```diff
@@ -166,11 +217,13 @@ type AgentCert struct {
 
 // Principals returns a map where the keys are the CA's PKS and the certs identities are the values
 func (a SSHAgent) Principals() ([]AgentCert, error) {
-	conn, err := net.Dial("unix", a.Socket)
+	conn, err := net.DialTimeout("unix", a.Socket, a.Timeout)
```

Comment:
Should this have a context plumbed through, use `DialContext` instead?

Reply:
"plumbed through" as passed by the caller? I think there's value in the caller not having to hard code a timeout or have its own flags / mechanics to configure a context, and just assume the library it is invoking enforces some sane behavior (can invoke the agent directly if something fancier is needed).

If you mean to just change a.Timeout to a context with deadline... I have no strong feelings around it. Either way it works. Let me know.